### PR TITLE
Use bootstrap-select in testplans/mutable.html

### PR DESCRIPTION
**Pull request by asankov**

### CODE REVIEW

**Comment by asankov** [comment]
> On page load the version select isn't updated anymore

Either I don't understand what you are saying or I can't reproduce it. Can you clarify what you mean?

> On Product change doesn't update version

This was a small issues of the way I declared `bootstrap-select` in the template. It had to be declared before `jQuery`. It is fixed now.

> When adding new Product via the + sign the value is displayed in the select but is not reflected in the drop down options. I can change to some of the previous values and I can't go back to the newly added value.

This one was a lot trickier. The thing is - `bootstrap-select` does not work how we expected it to. It does not update values automatically, and even when calling `.selectpicker('refresh')` it does not respect the `query` of the field. 
Before yourself you see the cleverest solution I managed to come up with after banging my head againts the keyboard for a few hours - refreshing the whole page. This is done via the `window.onuload` function which is called whenever the window is about to be closed. That is - the user creates a new product, clicks `Save`, the window is about to be closed and therefore the products are refreshed. The same goes for the product versions. 
However, there was one more problem - although `window.onuload` is supposed to be called when the window is about to be closed, for whatever reason the browser calls it immediately after loading. Having done so, the page is refreshed, and the `win` variable is no longer the same, which makes the whole function useless. So I had to implement the `if (isFirst)` check to prevent that initial reload.

IMO this is way too much work, and way too much code to be maintained for the future (and a lot of dirty code), just to use one stylish select. The alternative is to not use `bootstrap-select` here, and just be a little inconsistent in our templates. I'll be glad to hear your opinion on that.

**Comment by asankov** [comment]
Everything seems to be up and running now. Using `document.getElementById` rather than `$('#id')` solved all of our problems. 
Also `testruns/mutable.html` has been updated in the same manner

File: tcms/testplans/static/testplans/js/mutable.js [hunk 1]
```diff
@@ -10,6 +10,13 @@ $(document).ready((function () {
     $('#add_id_version').click(function () {
         return showRelatedObjectPopup(this);
     });
+
+    $('.selectpicker').selectpicker();
+
+    document.getElementById('id_product').onchange = function(){
+        $('.selectpicker').selectpicker('refresh');
+        populateProductVersion();
```
Comment (atodorov) [inline tcms/testplans/static/testplans/js/mutable.js]: This is correctly updating the Version select when a new Product is added. However it has the unwanted side effect that the first version is shown twice when changing between existing products. Test it more carefully next time.

Also:

1) The refresh action should be applied only to the product select
2) This new onchange handler duplicates the change handler so you have to remove `.change()`

3) Also missing the refresh for the Version select - again test it more carefully before submitting.

File: tcms/testruns/static/testruns/js/mutable.js [hunk 2]
```diff
@@ -4,4 +4,8 @@ $(document).ready(function() {
     });
 
     $('.selectpicker').selectpicker();
+
+    document.getElementById('id_build').onchange = function() {
+        $('.selectpicker').selectpicker('refresh');
```
Comment (atodorov) [inline tcms/testruns/static/testruns/js/mutable.js]: Apply only to id_build not all that match.

File: tcms/testplans/static/testplans/js/mutable.js (hunk 1 above)
Comment (asankov) [reply tcms/testplans/static/testplans/js/mutable.js]: the update for the Version select is done inside `populateProductVersion`

Comment (atodorov) [reply tcms/testplans/static/testplans/js/mutable.js]: Nope, it is not. When you + a new Version object the product field is not changed in any way and the Version select drop-down is not refreshed.

**Pull request merged**